File: solutions/javascript/word-count/1/word-count.js
const checkFirstAndLastChar = (array) => {
  for (let i = 0; i < array.length; i++) {
    if (array[i].startsWith("'")) {
        array[i] = array[i].slice(1);
    }
    if (array[i].endsWith("'")) {
        array[i] = array[i].slice(0, array[i].length - 1);
    }
  }
  return array;
}

export const countWords = (sentence) => {
  let sanitized = sentence.toLowerCase().split(/[^\w']+/);
  sanitized = checkFirstAndLastChar(sanitized).filter(Boolean);
  
  const words = new Set(sanitized.sort());

  const result = {};
  for (const word of words) {
      result[word] = 0;
  }

  for (const word in result) {
      for (let i = 0; i < sanitized.length; i++) {
          if (word === sanitized[i]) {
              result[word]++;
          }
      }
  }
  return result;
};

File: solutions/java/word-count/1/src/main/java/WordCount.java
import java.util.List;
import java.util.Map;
import java.util.TreeMap;

class WordCount {
    public Map<String, Integer> phrase(String input) {
        Map<String, Integer> count = new TreeMap<>();

        List<String> words = List.of(input.toLowerCase().split("[^a-zA-Z0-9']"));

        for (String word : words) {
            if (word.endsWith("'"))
                word = word.substring(0, word.length() - 1);
            if (word.startsWith("'"))
                word = word.substring(1);
            if (!word.isEmpty())
                count.merge(word, 1, Integer::sum);
        }
        return count;
    }
}
